addons/l10n_it_edi/models/account_move.py

```diff
@@ -125,18 +125,34 @@ def _compute_l10n_it_edi_is_self_invoice(self):
             We recognize these cases based on the taxes that target the VJ tax grids, which imply
             the use of VAT External Reverse Charge.
         """
-        it_tax_report_vj_lines = self.env['account.report.line'].search([
-            ('report_id.country_id.code', '=', 'IT'),
-            ('code', '=like', 'VJ%'),
-        ])
-        vj_lines_tags = it_tax_report_vj_lines.expression_ids._get_matching_tags()
-        for move in self:
-            if not move.is_purchase_document():
-                move.l10n_it_edi_is_self_invoice = False
-                continue
-            invoice_lines_tags = move.line_ids.tax_tag_ids
-            ids_intersection = set(invoice_lines_tags.ids) & set(vj_lines_tags.ids)
-            move.l10n_it_edi_is_self_invoice = bool(ids_intersection)
+        purchases = self.filtered(lambda m: m.is_purchase_document())
+        others = self - purchases
+        for move in others:
+            move.l10n_it_edi_is_self_invoice = False
+        if purchases:
```

Comment:
Yes, I wanted to avoid doing code for customer invoices.
Self-invoices are only for purchases, at the moment.
The only case could be TD27 that may be Miscellanous, but we can think of it later.
https://help.fattureincloud.it/help/articolo/651-crea-autofattura-elettronica-td26-td27